website/src/fieldsGenerator/connectionFields.js

```diff
@@ -3,127 +3,183 @@ import { getBoolean, getENVValue } from '../utils/commonUtils';
 export const connectionFields = (path, yamlData) => {
     const fields = [];
     const pathSplit = path.split('.')
-    const connections =  yamlData.core_lib.connections
-    const index = pathSplit.at(pathSplit.indexOf('connections')+1)
-    const connection = connections[index] 
+    const connections = yamlData.core_lib.connections
+    const index = pathSplit.at(pathSplit.indexOf('connections') + 1)
+    const connection = connections[index]
     const keyPrefix = `core_lib.connections.${index}`
     fields.push(
         {
-            title: "What is the name of the DB connection?",
-            type: "string",
+            title: 'What is the name of the connection?',
+            type: 'string',
             default_value: null,
             value: connection.key,
             mandatory: true,
             key: keyPrefix + '.key',
             // validatorCallback: validateFunc,
         },
         {
-            title: "Select DB connection",
-            type: "enum",
+            title: 'Select connection',
+            type: 'enum',
             mandatory: true,
-            value: connection["url"]["protocol"],
+            value: connection['config']['url']['protocol'].includes('http') ? 'solr' : connection['config']['url']['protocol'],
             // validatorCallback: validateFunc,
             options: [
-                "SQLite",
-                "Postgresql",
-                "MySQL",
-                "Oracle",
-                "MSSQL",
-                "Firebird",
-                "Sybase",
+                'SQLite',
+                'Postgresql',
+                'MySQL',
+                'Oracle',
+                'MSSQL',
+                'Firebird',
+                'Sybase',
+                'Solr',
+                'Neo4j'
             ],
-            key: keyPrefix + '.url.protocol',
-        }
+            key: keyPrefix + '.config.url.protocol',
```

Comment:
what changed here and in later rows. start with config??

Reply:
according to the new structure the url and other configurations will be under `config` so for that change

Reply:
```yaml
connections:
    - key: sqlconn
      migrate: false
      config_instantiate: true
      type: core_lib.connection.sql_alchemy_connection_registry.SqlAlchemyConnectionRegistry
      config:
        create_db: true
        log_queries: false
        session:
          pool_recycle: 3200
          pool_pre_ping: false
        url:
          protocol: postgresql
          username: ${oc.env:SQLCONN_USER}
          password: ${oc.env:SQLCONN_PASSWORD}
          host: ${oc.env:SQLCONN_HOST}
          port: ${oc.env:SQLCONN_PORT}
          file: ${oc.env:SQLCONN_DB}
```